dummy/modowners_test.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"bytes"
+	"io/fs"
+	"log"
+	"strings"
+	"testing"
+	"testing/fstest"
+)
+
+func TestCommonElement(t *testing.T) {
+	for _, test := range []struct {
+		A      []string
+		B      []string
+		Result bool
+	}{
+		{nil, nil, false},
+		{[]string{"a"}, []string{"a"}, true},
+		{[]string{"a", "b"}, []string{"a"}, true},
+		{[]string{"a"}, []string{"b"}, false},
+	} {
+		if hasCommonElement(test.A, test.B) != test.Result {
+			t.Error(test)
+		}
+	}
+}
+
+func TestCheckTable(t *testing.T) {
+	for _, test := range []struct {
+		fakeMod       fs.FS
+		logger        *log.Logger
+		args          []string
+		expectedError error
+	}{
+		{fstest.MapFS{"go.txd": &fstest.MapFile{Data: []byte(`
+		require (
+			cloud.google.com/go/storage v1.28.1 // @delivery
+			cuelang.org/go v0.5.0 // @as-code @backend-platform
+			github.com/Azure/azure-sdk-for-go v65.0.0+incompatible // indirect, @delivery
+			github.com/Masterminds/semver v1.5.0 // @delivery @backend-platform
+		)
+		`)}}, log.New(&bytes.Buffer{}, "", 0), []string{"go.txd"}, nil},
+	} {
+		if check(test.fakeMod, test.logger, test.args) != test.expectedError {
```

Comment:
Do we want just negative test cases or are we also asserting on the logger as well?  Youll usually have expectedResult and expectedErr and you can do something like:
```
err := check()

if expectedErr is not nil && expectedErr != err {
  t.Error(test) // I'd use assert here but will defer to serge
} else {
  logResults := reUsableFunction // to turn logs into your expected format
  if expectedResult != logResults {
    t.Error(test)
  }
}
```

Reply:
hm, i might need to check the logger as well. will look into this

Reply:
ok i added an if statement that checks the logger!